Automattic/wp-calypso
# Editor: Add back the setter for the sidebar open / close preference

@shaunandrews reported that he was seeing the sidebar behave differently for two different user accounts:

> It’s not remembering its state for either user. One is always open by default, and the other is always closed by default.

It looks like this got inadvertently left out during a [refactor](https://github.com/Automattic/wp-calypso/pull/19294/files#diff-914bdeefa512de6f78c9b7988546d6e3L276).

This adds explicit calls in the `openEditorSidebar` & `closeEditorSidebar` actions to persist the preference & removes the unused `setEditorSidebar` from the post editor component.

## To Test

* Run this branch
* Open the editor
* Toggle the editor sidebar
* Reload the page
  * You should see the same sidebar open / close state as before the reload

Repeat & confirm.
